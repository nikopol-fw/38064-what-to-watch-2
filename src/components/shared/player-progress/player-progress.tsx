import * as React from 'react';


interface Props {
  percentage: number;
  timingString: string;
}

export const PlayerProgress: React.FC<Props> = (props) => {
  const {percentage, timingString} = props;

  return (
    <div className="player__controls-row">
      <div className="player__time">
        <progress className="player__progress" value={percentage} max="100"/>
        <div className="player__toggler" style={{left: `${percentage}%`}}>Toggler</div>
      </div>
      <div className="player__time-value">{timingString}</div>
    </div>
  );
};
