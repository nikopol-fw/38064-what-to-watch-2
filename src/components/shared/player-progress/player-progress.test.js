import React from 'react';
import renderer from 'react-test-renderer';

import {PlayerProgress} from './player-progress';


const mock = {
  percentage: 50,
  timingString: `00:07:47`,
};

it(`PlayerProgress correctly renders afters relaunch`, () => {
  const tree = renderer.create(
      <PlayerProgress
        percentage={mock.percentage}
        timingString={mock.timingString}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
