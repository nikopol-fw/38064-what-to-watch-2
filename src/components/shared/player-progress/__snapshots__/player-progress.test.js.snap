// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerProgress correctly renders afters relaunch 1`] = `
<div
  className="player__controls-row"
>
  <div
    className="player__time"
  >
    <progress
      className="player__progress"
      max="100"
      value={50}
    />
    <div
      className="player__toggler"
      style={
        Object {
          "left": "50%",
        }
      }
    >
      Toggler
    </div>
  </div>
  <div
    className="player__time-value"
  >
    00:07:47
  </div>
</div>
`;
